package/MDAnalysis/coordinates/TNG.py

```diff
@@ -0,0 +1,507 @@
+means that all of the critical trajectory data (positions, box, velocities
+(if present), forces (if present)) must share the same stride in trajectory
+integrator steps. These critical blocks in the TNG file are henceforth called
+*special blocks*. Optional blocks (all blocks that are not special blocks)
+will not be read if they do not share an integrator step with, or are not
+divisible by the shared integrator step of the special blocks.
+
+
+References
+----------
+
+.. bibliography::
+    :filter: False
+    :style: MDA
+
+    Lundborg2014
+
+
+.. Links
+
+.. _GROMACS:
+    https://www.gromacs.org/
+.. _pytng:
+    https://github.com/MDAnalysis/pytng
+.. _pytng documentation:
+    https://www.mdanalysis.org/pytng/
+.. _TNG blocks:
+    https://www.mdanalysis.org/pytng/documentation_pages/Blocks.html
+.. _TNG specification:
+    https://gitlab.com/hugomacdermott/tng/-/blob/master/Trajectoryformatspecification.mk
+
+"""
+
+import warnings
+from typing import List, Optional
+
+import MDAnalysis as mda
+import numpy as np
+from MDAnalysis.coordinates import base
+from MDAnalysis.coordinates.timestep import Timestep
+from MDAnalysis.lib.mdamath import triclinic_box
+
+from ..due import Doi, due
+
+try:
+    import pytng
+except ImportError:
+    #: Indicates whether pytng is found.
+    HAS_PYTNG = False
+else:
+    #: Indicates whether pytng is found.
+    HAS_PYTNG = True
+
+
+class TNGReader(base.ReaderBase):
+    r"""Reader for the TNG format
+
+    The TNG format :cite:p:`Lundborg2014` is a format used in `GROMACS`_ for
+    storage of trajectory and topology information. This reader is are based on
+    the `pytng`_ package for reading TNG files. The reader is directed to the
+    `pytng documentation`_ and `TNG specification`_ for further reading.
+
+    The TNG format allows a wide range of compression
+    algorithms and unlike the compressed XTC format can also store velocities
+    and forces in addition to positions.
+
+    The contents of the *special blocks* (positions, box, velocities, forces)
+    are read into the timestep if present. Additional blocks are read into the
+    `ts.data` dictionary if they are available at the current frame.
+
+    .. versionadded:: 2.4.0
+    """
+
+    format = 'TNG'
+    # NOTE: Time units are in seconds unlike other GROMACS formats
+    units = {'time': 'second', 'length': 'nm', 'velocity': 'nm/ps',
+             'force': 'kJ/(mol*nm)'}
+
+    _box_blockname = "TNG_TRAJ_BOX_SHAPE"
+    _positions_blockname = "TNG_TRAJ_POSITIONS"
+    _velocities_blockname = "TNG_TRAJ_VELOCITIES"
+    _forces_blockname = "TNG_TRAJ_FORCES"
+    _special_blocks = [_box_blockname, _positions_blockname,
+                       _velocities_blockname, _forces_blockname]
+
+    @due.dcite(Doi("10.1002/jcc.23495"),
+               description="The TNG paper",
+               path=__name__)
+    def __init__(self, filename: str, convert_units: bool = True, **kwargs):
```

Comment:
Can you add the store_init_arguments decorator here please? (I _think_ we need it).